src/Controller/QuestionController.php

```diff
@@ -53,24 +60,55 @@ public function homepage(QuestionRepository $repository, int $page = 1)
     }
 
     /**
-     * @Route("/questions/new")
+     * @Route("/questions/new", name="app_question_new")
      * @IsGranted("ROLE_USER")
      */
-    public function new()
+    public function new(Request $request)
     {
-        // force login
-        // $this->denyAccessUnlessGranted('ROLE_USER'); // can throw an access denied exception
-        // if (!$this->isGranted('ROLE_ADMIN')) {
-        //     throw $this->createAccessDeniedException('No access for you'); // error viewed by devs
-        // }
+        $question = new Question();
+        // create a form:
+        $form = $this->createForm(QuestionFormType::class, $question);
+        $form->handleRequest($request);
+
+        // check if POST REQ
+        if ($form->isSubmitted() && $form->isValid()) {
+            // name and question are filled in at the form
+            $question->setOwner($this->getUser());
+
+            $this->em->persist($question);
+            $this->em->flush();
+            // do anything else:
+            // flash a success message to the user
+            $this->addFlash('success', 'Your Question has been saved!');
+            // redirect to questions show page
+            return $this->redirectToRoute('app_question_show', [
+                'slug'=>$question->getSlug(),
+            ]);
+        }
+        // no submittion, initial render 
+        return $this->render('questions/new.html.twig', [
+            'questionForm' => $form->createView(),
+        ]);
+    }
+
+    #[Route('/questions/publish/{slug}', name: "app_question_publish")]
+    public function publish(Question $question)
+    {
+        $dt = new DateTime();
+        $question->setAskedAt($dt);
+
+        $this->em->persist($question);
```

Comment:
You don't need to persist this. You only need that when you are creating a new instance - not when it exists already.

Reply:
Done